xarray/tests/test_dtypes.py

```diff
@@ -50,3 +50,47 @@ def error():
 def test_inf(obj):
     assert dtypes.INF > obj
     assert dtypes.NINF < obj
+
+
+@pytest.mark.parametrize("kind, expected", [
+    ('a', (np.dtype('O'), np.nan)),  # dtype('S')
+    ('b', (np.float32, np.nan)),  # dtype('int8')
+    ('B', (np.float32, np.nan)),  # dtype('uint8')
+    ('c', (np.dtype('O'), np.nan)),  # dtype('S1')
+    ('D', (np.complex128, np.complex('nan+nanj'))),  # dtype('complex128')
+    ('d', (np.float64, np.nan)),  # dtype('float64')
+    ('e', (np.float16, np.nan)),  # dtype('float16')
+    ('F', (np.complex64, np.complex('nan+nanj'))),  # dtype('complex64')
+    ('f', (np.float32, np.nan)),  # dtype('float32')
+    ('G', (np.complex256, np.complex('nan+nanj'))),  # dtype('complex256')
+    ('g', (np.float128, np.nan)),  # dtype('float128')
```

Comment:
Apparently `complex256` and `float128` aren't always available -- that's why our integration tests on Appveyor (running Windows) fail for this PR. So let's exclude these.

Reply:
I ignored float128 (and complex256, see https://github.com/pydata/xarray/pull/1953#discussion_r210686841)